expr/random.go

```diff
@@ -69,10 +70,13 @@ func NewRandom(seed string) *ExampleGenerator {
 type ExampleGenerator struct {
 	Randomizer
 	seen map[string]*any
+	mu   sync.RWMutex
```

Comment:
While this isn't essential right now, I think it's sensible to add locks around map operations to avoid panics in future!